src/features/todos.ts

```diff
@@ -1,8 +1,23 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
 import { Todo } from '../types/Todo';
 
+type StateType = {
+  todos: Todo[];
+};
+
+export const initialState: StateType = {
+  todos: [],
+};
+
 export const todosSlice = createSlice({
   name: 'todos',
-  initialState: [] as Todo[],
-  reducers: {},
+  initialState,
+  reducers: {
+    setTodos: (state, action: PayloadAction<Todo[]>) => {
+      // eslint-disable-next-line no-param-reassign
+      state.todos = action.payload;
```

Comment:
You shouldn't reassign or mutate state in Redux. Always use this syntax to ensure immutability. It's safer, especially if this slice could handle other properties in the future

```
return {
        ...state,
        todos: action.payload,
 };
```